src/meta/src/manager/catalog/mod.rs

```diff
@@ -324,6 +335,40 @@ where
         }
     }
 
+    /// Each connection is identified by a unique name
+    pub async fn create_connection(
+        &self,
+        connection: Connection,
+    ) -> MetaResult<NotificationVersion> {
+        let core = &mut self.core.lock().await.connection;
+        core.check_connection_duplicated(&connection.name)?;
+
+        let conn_id = connection.id;
+        let conn_name = connection.name.clone();
+        let mut connections = BTreeMapTransaction::new(&mut core.connections);
+        connections.insert(conn_id, connection);
+        commit_meta!(self, connections)?;
+
+        core.connection_by_name.insert(conn_name, conn_id);
+        // Currently we don't need to notify frontend, so just fill 0 here
+        Ok(0)
+    }
+
+    pub async fn drop_connection(&self, conn_name: &str) -> MetaResult<NotificationVersion> {
+        let core = &mut self.core.lock().await.connection;
+
+        let conn_id = core
+            .connection_by_name
+            .remove(conn_name)
+            .ok_or_else(|| anyhow!("connection {} not found", conn_name))?;
+
+        let mut connections = BTreeMapTransaction::new(&mut core.connections);
+        connections.remove(conn_id);
+        commit_meta!(self, connections)?;
+        // Currently we don't need to notify frontend, so just fill 0 here
```

Comment:
If the endpoint creation doesn't approve by the user, create source will fail since we cannot connect to Kafka. If we wait in the create connection process, the user may feel confused that what happens.

Reply:
I mean if users don't approve, enumerators can never access the broker, and create source process can fail. 
& what if creating a table with kafka connector using private link